src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -259,6 +259,49 @@ private static SyntaxNode WithBlockBody(SyntaxNode node, BlockSyntax body)
             return document.Document.WithSyntaxRoot(newRoot);
         }
 
+        private int GetFirstStatementAffectedIndex(SyntaxNode innermostCommonBlock, ISet<ExpressionSyntax> matches, int firstStatementAffectedIndex)
+        {
+            // If a local function is involved, we have to make sure the new declaration is placed:
+            //     1. Before all calls to local functions that use the variable.
+            //     2. Before the local function(s) themselves.
+            //     3. Before all matches.
+            // Cases (2) and (3) are already covered by the 'firstStatementAffectedIndex' parameter. Thus, all we have to do is ensure we consider (1) when
+            // determining where to place our new declaration.
+
+            // Find all the local functions within the scope that will use the new declaration.
+            var localFunctions = innermostCommonBlock.DescendantNodes().Where(node => node.IsKind(SyntaxKind.LocalFunctionStatement) && matches.Any(match => match.Span.OverlapsWith(node.Span)));
+
+            if (localFunctions.IsEmpty())
+            {
+                return firstStatementAffectedIndex;
+            }
+
+            var localFunctionIdentifiers = localFunctions.Select(node => ((LocalFunctionStatementSyntax)node).Identifier.ValueText);
+
+            // Find all calls to the applicable local functions within the scope.
+            var localFunctionCalls = innermostCommonBlock.DescendantNodes().Where(node => node is InvocationExpressionSyntax invocationExpression &&
+                                                                                  localFunctionIdentifiers.Contains(invocationExpression.Expression.GetRightmostName().Identifier.ValueText));
```

Comment:
this is inaccurate.  For example, `this.localName()` will show up for you as a call to a local function `localName` using this logic.

Reply:
`GetRightmostName` can return null in lots of legal cases.  for example `(foo)(arg1, arg2)`  this will crash.

Reply:
Got it, thank you for the catch! Updated. 👍